### DISCUSSION TITLE
Does this sample work for editprofile?

### DISCUSSION THREAD

**Discussion by hilanderjeff** [Azure-Samples/ms-identity-python-webapp#92] [General]
Hi,

I tried the example. But the latest version doesn't work with resetpassword and editprofile. I wonder the authorized URL is not correct. This is the URL I constructed. https://dev.login.firstam.com/myidaasdev.onmicrosoft.com/B2C_1A_FA_EIDv2_ProfileEdit?client_id=myclientid. This is not an auth uri. Will it work?

Thanks

**Comment by rayluo** [comment]
So, you are using this sample for your B2C tenant? How did you configure this sample? There is a dedicated [README_B2C](https://github.com/Azure-Samples/ms-identity-python-webapp/blob/master/README_B2C.md) which should automatically construct a correct edit profile URL for you.

**Comment by hilanderjeff** [comment]
I do follow the instructions to configure the sample. I got signin working
with my custom policy but the constructed edit profile URL did not work for
me. Also the reset password neither worked.

**Reply by rayluo** [reply]
That URL in your first message is indeed unreachable. Perhaps you should go through some more docs on the B2C side to make sure they are working, before attempting this sample? Sorry we (the maintainers of this sample) have no more specific answer than this.

**Comment by hilanderjeff** [comment]
Got it. Thank you for your response.
BTW, do you have similar sample like this one with MSAL library directly?

**Reply by rayluo** [reply]
[MSAL Python has many samples, organized by scenarios](https://msal-python.readthedocs.io/en/latest/). And this sample that you are currently using *IS* the sample for web app. In its earlier version, it indeed uses MSAL Python library directly, but the latest version simplifies a lot of things. We would suggest you to start/stay with this latest sample.
